app/views/incomeTaxSummaryIncomeSourceDetail.scala.html

```diff
@@ -1,79 +1,119 @@
-                        @link(
-                            id = Some(s"${typeLabel}${idx}TaxCodeLink"),
-                            url = taxCodeUrl.url,
-                            copy = Messages("tai.incomeTaxSummary.taxCode.prefix", incomeSource.taxCode),
-                            linkClasses = Seq("display-tax-code-for-print"))
-                    }
-                }
-                @if(incomeSource.displayPayrollNumber && typeLabel=="employment"){
-                    <span id="@{typeLabel}@{idx}PayeNumber" class="carry-over govuk-!-display-block display-block ">@messages("tai.income.details.ERN"):@incomeSource.taxDistrictNumber/@incomeSource.payeNumber</span>
-                }
-
-                @if(incomeSource.displayPayrollNumber && typeLabel == "pension"){
-                <span id="@{typeLabel}@{idx}PensionNumber" class="carry-over govuk-!-display-block display-block ">@Messages("tai.incomeTaxSummary.pensionNumber.prefix", incomeSource.payrollNumber)</span>
-                }
-
-                @if(incomeSource.displayEndDate){
-                    <span id="@{typeLabel}@{idx}EndDate" class="govuk-!-display-block display-block">@Messages("tai.incomeTaxSummary.endDate.prefix", incomeSource.endDate)</span>
-                }
-
-            </p>
-
-            @if(incomeSource.displayDetailsLink){
-            <p class="govuk-body flush--top">
-                @link(
-                    id=Some(s"${typeLabel}${idx}DetailsLink"),
-                    url=incomeSource.detailsLinkUrl,
-                    copy=incomeSource.detailsLinkLabel,
-                    altCopy=Some(Messages("tai.updateOrRemove.fromOtherSources", incomeSource.name))
-                )
-            </p>
-            }
+
+<div class="govuk-grid-row grid-row">
+
+    @for((incomeSource, idx) <- pair zip (Stream from pidx)) {
+    <div id="@typeLabel@idx"
+         class="subsection flush--top @if(pair.length==1){column-full govuk-grid-column-full}else{column-one-half govuk-grid-column-one-half}">
+
+        <h3 id="@{typeLabel}@{idx}Heading" class="govuk-heading-l heading-large flush break-word-only">
+            @incomeSource.name</h3>
+
+        <p class="govuk-body govuk-!-font-size-36 govuk-!-font-weight-bold faded-text-dark flush--top ">
+            <span class="govuk-visually-hidden ">@accessibleAmountPrefix </span>
+            <span class="faded-text-dark flush--top">@incomeSource.amount</span>
+        </p>
+
+        <p class="govuk-body flush--top">
+            @if(incomeSource.displayTaxCode) {
+
+            @incomeSource.taxCodeUrl.map { taxCodeUrl =>
+            @link(
+            id = Some(s"${typeLabel}${idx}TaxCodeLink"),
+            url = taxCodeUrl.url,
+            copy = Messages("tai.incomeTaxSummary.taxCode.prefix", incomeSource.taxCode),
+            linkClasses = Seq("display-tax-code-for-print"))
+        <div aria-hidden="true" style="display: none;" id="taxCodeList1">
```

Comment:
should. the hidden div id=taxCodeList1  not appear after the closing P tag
This might allow Tim to toggle ie  hide the existing P tag when the div is shown

Reply:
also use `idx` rather than 1 or we'll get the same index repeated many times

Reply:
this makes sense, I will make the change

Reply:
only issue is there is another div in the file with id="taxCodeList1" 

` <div aria-hidden="true" style="display: none;" **id="taxCodeList1"**>
                <details class="govuk-details" data-module="govuk-details">
                    <summary class="govuk-details__summary">
                                <span class="govuk-details__summary-text">
                                  @Messages("tai.incomeTaxSummary.taxCode.prefix", incomeSource.taxCode)
                                </span>
                    </summary>
                    <div class="govuk-details__text">

                        <div class="govuk-!-padding-bottom-3">
                            @for((taxCodeList, listIdx) <-
                            taxCodeViewModel.get(incomeSource.taxCode).map(_.taxCodeDetails).getOrElse(Seq.empty[DescriptionListViewModel])
                            zip (Stream from 1)) {
                            <div id=**"taxCodeList@{listIdx.toString}"**>`

Reply:
I had to make a significant change because there was two instances of id="taxCodeList@{Idx}", please see latest commit